lib/matplotlib/backend_tools.py

```diff
@@ -99,20 +98,6 @@ def trigger(self, sender, event, data=None):
 
         pass
 
-    @figure.setter
-    def figure(self, figure):
-        """
-        Set the figure
-
-        Set the figure to be affected by this tool
-
-        Parameters
-        ----------
-        figure: `Figure`
-        """
-
-        self._figure = figure
```

Comment:
Ahh yes, I understand, sorry, getting confused as to where bits of code lie.  Yes, I agree we shouldn't have this setter.  For some reason I thought we had this bit of code in ``ToolManager``.

Yes, I completely agree we want to convert this setter in ``ToolBase`` into just a callback that sets the figure.

Going back, I think I got confused because I saw it completely logical to have a figure setter in ``ToolManager`` and instead of passing in a ``Canvas`` or a ``FigureManager`` into ``ToolManager`` we just pass in a ``figure``.